tests/director/tests/formation_api_test.go

```diff
@@ -194,15 +189,64 @@ func TestApplicationFormationFlow(t *testing.T) {
 	saveExampleInCustomDir(t, unassignReq.Query(), unassignFormationCategory, "unassign application from formation")
 
 	t.Log("Should be able to delete formation after application is unassigned")
-	deleteRequest = fixtures.FixDeleteFormationRequest(newFormation)
-	var deleteFormation graphql.Formation
-	err = testctx.Tc.RunOperationWithCustomTenant(ctx, certSecuredGraphQLClient, tenantId, deleteRequest, &deleteFormation)
-	assert.NoError(t, err)
-	assert.Equal(t, newFormation, deleteFormation.Name)
+	fixtures.DeleteFormationWithinTenant(t, ctx, certSecuredGraphQLClient, tenantId, newFormation)
 
 	saveExample(t, deleteRequest.Query(), "delete formation")
 }
 
+func TestApplicationOnlyFormationFlow(t *testing.T) {
+	// GIVEN
+	ctx := context.Background()
+	newFormation := "ADDITIONAL"
+
+	tenantId := tenant.TestTenants.GetDefaultTenantID()
+	subaccountID := tenant.TestTenants.GetIDByName(t, tenant.TestProviderSubaccount)
+
+	t.Log("Create formation template")
+	input := graphql.FormationTemplateInput{Name: "application-only-formation-template", ApplicationTypes: []string{string(util.ApplicationTypeC4C)}}
+	formationTemplate := fixtures.CreateFormationTemplate(t, ctx, certSecuredGraphQLClient, input)
+	defer fixtures.CleanupFormationTemplate(t, ctx, certSecuredGraphQLClient, formationTemplate.ID)
+
+	t.Logf("Should create formation: %s", newFormation)
+	fixtures.CreateFormationFromTemplateWithinTenant(t, ctx, certSecuredGraphQLClient, tenantId, newFormation, &formationTemplate.Name)
+	defer fixtures.DeleteFormation(t, ctx, certSecuredGraphQLClient, newFormation)
+
+	t.Log("Create application")
+	app, err := fixtures.RegisterApplicationWithApplicationType(t, ctx, certSecuredGraphQLClient, "app", conf.ApplicationTypeLabelKey, string(util.ApplicationTypeC4C), tenantId)
+	defer fixtures.CleanupApplication(t, ctx, certSecuredGraphQLClient, tenantId, &app)
+	require.NoError(t, err)
+	require.NotEmpty(t, app.ID)
+
+	formationInput := graphql.FormationInput{Name: newFormation, TemplateName: &formationTemplate.Name}
+
+	t.Logf("Assign application to formation %s", newFormation)
+	defer fixtures.UnassignFormationWithApplicationObjectType(t, ctx, certSecuredGraphQLClient, formationInput, app.ID, tenantId)
+	fixtures.AssignFormationWithApplicationObjectType(t, ctx, certSecuredGraphQLClient, formationInput, app.ID, tenantId)
+
+	t.Logf("Create runtime")
+	rtmName := "rt"
+	rtmInput := fixRuntimeInput(rtmName)
+
+	runtime := fixtures.RegisterKymaRuntime(t, ctx, certSecuredGraphQLClient, subaccountID, rtmInput, conf.GatewayOauth)
+	defer fixtures.CleanupRuntime(t, ctx, certSecuredGraphQLClient, subaccountID, &runtime)
+
+	t.Logf("Should fail to assign runtime to formation %s", newFormation)
+	defer fixtures.UnassignFormationWithRuntimeObjectType(t, ctx, certSecuredGraphQLClient, formationInput, runtime.ID, tenantId)
+	fixtures.AssignFormationWithRuntimeObjectTypeExpectError(t, ctx, certSecuredGraphQLClient, formationInput, runtime.ID, tenantId)
```

Comment:
I think it's worth it asserting the error message (or that it contains specific part so that it can be unified/reused) as well not only that there is an error because from my PoV it is important to know that we received the concrete error in this flow.

What do you think?

Reply:
As discussed offline, for now we are leaving the tests this way